File: 백준/10828.py
import sys

stack=[]

N=int(sys.stdin.readline())
for i in range(N):
    str=sys.stdin.readline().split()

    if str[0]=="push":
        stack.append(str[1])
    elif str[0] =="pop":
        if len(stack)==0:
            print(-1)
        else:
            print(stack.pop())
    elif str[0] =="size":
        print(len(stack))
    elif str[0] =="empty":
        if len(stack)==0:
            print(1)
        else:
            print(0)
    elif str[0] =="top":
        if len(stack)==0:
            print(-1)
        else:
            print(stack[-1])

File: 2022/백준/cpp/10828.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    int n;
    cin >> n;

    stack<int> s;

    for(int i=0;i<n; i++){
        string input;
        cin >> input;
        if(input == "push"){
            int x;
            cin >> x;
            s.push(x);
        }else if(input == "pop"){
            if(s.empty()){
                cout << -1 << endl;
            }else{
                cout << s.top() <<endl;
                s.pop();
            }
        }else if(input == "size"){
            cout << s.size() <<endl;
        }else if(input == "empty"){
            if(s.empty()){
                cout << 1 <<endl;
            }else{
                cout << 0 <<endl;
            }
        }else if(input == "top"){
            if(s.empty()){
                cout << -1 << endl;
            }else{
                cout << s.top() <<endl;
            }
        }
    }

    return 0;
}